extensions/SharkPool/Recording.js

```diff
@@ -0,0 +1,285 @@
+      this.audioRecorder = null;
+      this.analyzerNode = null;
+    }
+
+    getInfo() {
+      return {
+        id: "SPrecording",
+        name: "Recording",
+        color1: "#7B9149",
+        color2: "#64753C",
+        color3: "#505E30",
+        menuIconURI,
+        blockIconURI,
+        blocks: [
+          {
+            opcode: "recordingSet",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "recording mode [MODE]",
+            arguments: {
+              MODE: { type: Scratch.ArgumentType.STRING, menu: "MODE" }
+            }
+          },
+          {
+            opcode: "recordForX",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "record for [TIME] seconds",
+            arguments: {
+              TIME: { type: Scratch.ArgumentType.NUMBER, defaultValue: 1 }
+            }
+          },
+          {
+            opcode: "clearRecording2",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "clear recording"
+          },
+          {
+            opcode: "averageLoudness",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "average loudness"
+          },
+          "---",
+          {
+            opcode: "whenMic",
+            blockType: Scratch.BlockType.EVENT,
+            text: "when mic turns [ON_OFF]",
+            isEdgeActivated: false,
+            arguments: {
+              ON_OFF: { type: Scratch.ArgumentType.STRING, menu: "ACTIVE" }
+            }
+          },
+          {
+            opcode: "isRecordingMic",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "is recording?"
+          },
+          "---",
+          {
+            opcode: "recordedAudio",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "recorded audio as [TYPE]",
+            arguments: {
+              TYPE: { type: Scratch.ArgumentType.STRING, menu: "FILETYPES" }
+            }
+          },
+          {
+            opcode: "saveRecording",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "save recording to [SPRITE] named [NAME]",
+            arguments: {
+              SPRITE: { type: Scratch.ArgumentType.STRING, menu: "TARGETS" },
+              NAME: { type: Scratch.ArgumentType.STRING, defaultValue: "Recording 1" }
+            }
+          },
+          {
+            opcode: "saveRecording2",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "download recording named [NAME] as [TYPE]",
+            arguments: {
+              NAME: { type: Scratch.ArgumentType.STRING, defaultValue: "Recording_1" },
+              TYPE: { type: Scratch.ArgumentType.STRING, menu: "FILETYPES" }
+            }
+          }
+        ],
+        menus: {
+          TARGETS: { acceptReporters: true, items: "_getTargets" },
+          ACTIVE: ["on", "off"],
+          MODE: { acceptReporters: true, items: ["enabled", "disabled"] },
+          FILETYPES: {
+            acceptReporters: true,
```

Comment:
```suggestion
            acceptReporters: false,
```
It's probably not a good idea to accept reporters here, since the way you've programmed the function doesn't check whether a file extension actually exists or not. I didn't personally run into any bugs with this, however I imagine this could cause some unintended functionality.